src/TideCoreOperation.php

```diff
@@ -233,4 +234,76 @@ public function chagneDiffSettings() {
     }
   }
 
+  /**
+   * Enable TFA.
+   */
+  public static function enableTfaNecessaryModules() {
+    $moduleHandler = \Drupal::service('module_handler');
+    $moduleInstaller = \Drupal::service('module_installer');
+    // Enable Real AES.
+    if (!$moduleHandler->moduleExists('real_aes')) {
+      $moduleInstaller->install(['real_aes']);
+    }
+    // Enable Two-factor Authentication (TFA).
+    if (!$moduleHandler->moduleExists('tfa')) {
+      $moduleInstaller->install(['tfa']);
+    }
+  }
+
+  /**
+   * Disabled site alert for TFA routes.
+   */
+  public static function disabledSiteAlertTfa() {
+    $block = Block::load('tide_site_alert_header');
+    if (\Drupal::service('module_handler')->moduleExists('tide_site_alert') && $block) {
+      $block->setVisibilityConfig('request_path', [
+        'id' => 'request_path',
+        'pages' => "/user/*/security/tfa\n/user/*/security/tfa/*",
+        'negate' => TRUE,
+      ]);
+      $block->save();
+    }
+  }
+
+  /**
+   * Update TFA settings.
+   */
+  public static function updateTfaSettings(array $config_install, array $config_optional) {
+    \Drupal::moduleHandler()->loadInclude('tide_core', 'inc', 'includes/helpers');
+    $configs_files_install = [
+      'key.key.tfa_encryption_key',
+      'encrypt.profile.tfa_encryption',
+    ];
+
+    $config_files_optional = [
+      'encrypt.settings',
+      'tfa.settings',
+    ];
+
+    foreach ($configs_files_install as $install) {
+      _tide_ensure_config($install, $config_install);
```

Comment:
This may not be related to this PR, but I think using `_tide_ensure_config` breaks the uuid for configurations. We should avoid abruptly overwriting its upstream configurations.

cc @anthony-malkoun / @krakerag

Reply:
I haven't used _tide_ensure_config but it does seem to forcibly reinstall a config, which would generate a fresh uuid every time.

It's probably great for brand new configuration, but if you're updating existing I think we have the paradigm in a lot of our libraries about using _tide_read_config and checking for load and updating that way?

Reply:
_tide_ensure_config uses `\Drupal::service('config.storage')->write($id, $config_data);` This type of writing cannot generate a new UUID. It simply writes the YAML file to the database.

> It's probably great for brand new configuration, but if you're updating existing I think we have the paradigm in a lot of our libraries about using _tide_read_config and checking for load and updating that way?
yes, @krakerag, we have to read the configurations to a variable and then let Drupal handle the rest.

and in this case, we should not define `encrypt.settings.yml` and `tfa.settings.yml` here since they are already defined by tide_core's upstream. @yeniatencio

Reply:
@vincent-gao @krakerag  I have updated the code to use _tide_read_config for installing the encryption key and profile. 
Regarding `encrypt.settings.yml` and `tfa.settings.yml`, I still using _tide_ensure_config function as those files do not have a uuid. I need to defined those files and update them later on the hook update as the TFA module (when enable) will generate them with some basic values. It seems is not possible to add them on the array and use the same _tide_read_config as is not picking up simple configuration types.

Reply:
@yeniatencio This is not the correct approach. Removing default configurations and directly adding new ones breaks the change tracking mechanism.

Reply:
hey @yeniatencio ,
I think you could modify the default configurations instead of overwriting them. In my opinion, this change should be a submodule of tide_core, and there should be a mechanism to `disable` or `turn off` the module, especially considering that this is a significant change for the SDP. This way, in case something goes wrong, you would have the option to disable it without breaking anything.
cc @krakerag

Reply:
@vincent-gao,
The default configuration gets saved in the db once the TFA module is enabled. are you saying to call a function to set the value of each field? isn't the same thing as overwriting them? Not really sure what you mean or if I'm misunderstanding how to do it. 
The TFA module already provides the drush commands to disable/enable the module: 

- Disable TFA with `drush config-set tfa.settings enabled 0`
- Enable TFA with `drush config-set tfa.settings enabled 1`

Reply:
1. move all of your changes to a submodule, which can be named tide_tfa, under tide_core
2. remove `config/optional/encrypt.settings.yml`  and `config/optional/tfa.settings.yml`
3. move `config/install/encrypt.profile.tfa_encryption.yml` and `config/install/key.key.tfa_encryption_key.yml` to the `optional` directory.

Yes, they are different. Overwriting using `write` function will break `uuid` or `default_config_hash`, we should avoid this kind of operation.

Reply:
Hey @vincent-gao, I have updated my pr for your review. Thanks you for the explanation yesterday.